libs/coin-modules/coin-bitcoin/src/wallet-btc/storage/index.ts

```diff
@@ -126,31 +126,56 @@ class BitcoinLikeStorage implements IStorage {
 
       tx.outputs.forEach(output => {
         if (output.address === tx.address) {
+          log(
+            "bitcoin[storage]",
+            `Adding unspent output: ${output.output_hash}:${output.output_index} -> ${tx.address}`,
+          );
           this.unspentUtxos[indexAddress].push(output);
         }
       });
 
       tx.inputs.forEach(input => {
-        if (input.address === tx.address) {
-          this.spentUtxos[indexAddress].push(input);
-        }
+        const inputAddress = input.address;
+
+        // Initialize arrays if needed
+        this.spentUtxos[inputAddress] = this.spentUtxos[inputAddress] || [];
+        this.unspentUtxos[inputAddress] = this.unspentUtxos[inputAddress] || [];
```

Comment:
```suggestion
        this.spentUtxos[inputAddress] = this.spentUtxos[inputAddress] ?? [];
        this.unspentUtxos[inputAddress] = this.unspentUtxos[inputAddress] ?? [];
```
We are dealing with arrays, can't be `0` or `""`